components/ca-treeview.js

```diff
@@ -0,0 +1,263 @@
+define(
+['/components/helpers/create-element.js', '/components/helpers/parent-by-attribute.js', '/components/helpers/cancel-event.js', 'document-register-element'],
+(createElement, parentByAttribute, cancelEvent) => {
+    /**
+     * Class TreeView for a TreeView web component
+     * @extends HTMLElement
+     */
+    class TreeView extends HTMLElement {
+        /**
+         * createdCallback is called when the component is first created.
+         * creates children, binds events and loads data if src is avaliable.
+         * @returns {undefined} nothing.
```

Comment:
Yeah, for some reason, the linter errors if I omit the `@returns` - I don't like having a @returns if there is no return, it's very annoying.